server/protocols/webadmin/webadmin-cassandra-data/src/main/java/org/apache/james/webadmin/service/CassandraMappingsSolveInconsistenciesTask.java

```diff
@@ -0,0 +1,57 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.service;
+
+import javax.inject.Inject;
+
+import org.apache.james.rrt.cassandra.CassandraMappingsSourcesDAO;
+import org.apache.james.rrt.cassandra.migration.MappingsSourcesMigration;
+import org.apache.james.task.Task;
+
+import reactor.core.publisher.Mono;
+
+public class CassandraMappingsSolveInconsistenciesTask implements Task {
+    public static final String TYPE = "cassandraMappingsSolveInconsistencies";
+
+    private final MappingsSourcesMigration mappingsSourcesMigration;
+    private final CassandraMappingsSourcesDAO cassandraMappingsSourcesDAO;
+
+    @Inject
+    CassandraMappingsSolveInconsistenciesTask(MappingsSourcesMigration mappingsSourcesMigration,
+                                                     CassandraMappingsSourcesDAO cassandraMappingsSourcesDAO) {
+        this.mappingsSourcesMigration = mappingsSourcesMigration;
+        this.cassandraMappingsSourcesDAO = cassandraMappingsSourcesDAO;
+    }
+
+    @Override
+    public Result run() {
+        return cassandraMappingsSourcesDAO.truncateTable()
+            .then(Mono.just(mappingsSourcesMigration.run()))
```

Comment:
be careful with `.then()`, transforming operations (`.map()`, `.flatMap()`, ...) will not be called after `.then()` because `.then()`  returns a `Mono<Void>`, there is no value in a Mono<Void>, therefore you cant use map.
I guess your `run()` method will return a `null` Result even when it is success. Please check it

Be careful, Having  a mocking statements in your test hide actual behavior of this task:
```java
when(mappingsSourcesMigration.run()).thenReturn(Task.Result.COMPLETED);
```

Reply:
I think I disagree with that sorry. If it's `then()` yes it returns `Mono<Void>`, if it's `then(Mono<R>)` it seems to return a `Mono<R>` (here `Mono<Result>`). 

Taking out the mocks so I could see it works as expected.